mesonbuild/dependencies/dev.py

```diff
@@ -534,8 +534,11 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
         modules: T.List[str] = mesonlib.listify(kwargs.get('modules', []))
         for module in modules:
             if module not in {'jvm', 'awt'}:
-                log = mlog.error if self.required else mlog.debug
-                log(f'Unknown JNI module ({module})')
+                msg = f'Unknown JNI module ({module})'
+                if self.required:
+                    mlog.error(msg)
+                else:
+                    mlog.debug(msg)
```

Comment:
> There should be a way to make mypy happy without casting, but I can't figure it out, since the mlog.error and mlog.debug actually have different signatures.

Hmm, mypy still fails, because in commit 7254db36a7adfa3f5e3e6cbce5dc1e3037893eb9 we added more `log = mlog.error if self.required else mlog.debug`.